lustre_exporter_test.go

```diff
@@ -0,0 +1,1362 @@
+		{"lustre_disk_io", "Current number of I/O operations that are processing during the snapshot.", gauge, []labelPair{{"operation", "read"}, {"ost", "lustrefs-OST0000"}, {"size", "3"}}, 0, false},
+		{"lustre_disk_io", "Current number of I/O operations that are processing during the snapshot.", gauge, []labelPair{{"operation", "read"}, {"ost", "lustrefs-OST0000"}, {"size", "4"}}, 0, false},
+		{"lustre_disk_io", "Current number of I/O operations that are processing during the snapshot.", gauge, []labelPair{{"operation", "read"}, {"ost", "lustrefs-OST0000"}, {"size", "5"}}, 0, false},
+		{"lustre_disk_io", "Current number of I/O operations that are processing during the snapshot.", gauge, []labelPair{{"operation", "read"}, {"ost", "lustrefs-OST0000"}, {"size", "6"}}, 0, false},
+		{"lustre_disk_io", "Current number of I/O operations that are processing during the snapshot.", gauge, []labelPair{{"operation", "read"}, {"ost", "lustrefs-OST0000"}, {"size", "7"}}, 0, false},
+		{"lustre_disk_io", "Current number of I/O operations that are processing during the snapshot.", gauge, []labelPair{{"operation", "read"}, {"ost", "lustrefs-OST0000"}, {"size", "8"}}, 0, false},
+		{"lustre_disk_io", "Current number of I/O operations that are processing during the snapshot.", gauge, []labelPair{{"operation", "read"}, {"ost", "lustrefs-OST0000"}, {"size", "9"}}, 0, false},
+		{"lustre_disk_io", "Current number of I/O operations that are processing during the snapshot.", gauge, []labelPair{{"operation", "write"}, {"ost", "lustrefs-OST0000"}, {"size", "1"}}, 3.9683886e+07, false},
+		{"lustre_disk_io", "Current number of I/O operations that are processing during the snapshot.", gauge, []labelPair{{"operation", "write"}, {"ost", "lustrefs-OST0000"}, {"size", "10"}}, 10445, false},
+		{"lustre_disk_io", "Current number of I/O operations that are processing during the snapshot.", gauge, []labelPair{{"operation", "write"}, {"ost", "lustrefs-OST0000"}, {"size", "11"}}, 1172, false},
+		{"lustre_disk_io", "Current number of I/O operations that are processing during the snapshot.", gauge, []labelPair{{"operation", "write"}, {"ost", "lustrefs-OST0000"}, {"size", "12"}}, 254, false},
+		{"lustre_disk_io", "Current number of I/O operations that are processing during the snapshot.", gauge, []labelPair{{"operation", "write"}, {"ost", "lustrefs-OST0000"}, {"size", "13"}}, 47, false},
+		{"lustre_disk_io", "Current number of I/O operations that are processing during the snapshot.", gauge, []labelPair{{"operation", "write"}, {"ost", "lustrefs-OST0000"}, {"size", "14"}}, 10, false},
+		{"lustre_disk_io", "Current number of I/O operations that are processing during the snapshot.", gauge, []labelPair{{"operation", "write"}, {"ost", "lustrefs-OST0000"}, {"size", "2"}}, 4.204676e+06, false},
+		{"lustre_disk_io", "Current number of I/O operations that are processing during the snapshot.", gauge, []labelPair{{"operation", "write"}, {"ost", "lustrefs-OST0000"}, {"size", "3"}}, 7
```

Comment:
Sadly, I followed a fairly manual process to create them. I added every file we are parsing in `/proc/fs/lustre` and `/proc/sys/lnet` and edited the tests in `lustre_exporter_test.go` to spit out every metric that it parsed but wasn't in this list. Then I just hacked together a regex script that converted the golang struct into this type. Very tedious process, but it should be much easier in the future assuming any new additions will be of a smaller scale.

As a follow-up though, I tried thinking of a way to verify that each metric was used, but nothing was readily obvious. If we could somehow read the metrics in `procfs` and `procsys` and make sure it is in the list, I think that would be useful. Could be I'm off-base there or it won't add much value, but that was my thought.